### PR TITLE
Snapper cleanup test based on fate#312751

### PR DESCRIPTION
**Pull request by dgutu-suse**
Test case based on fate, real example of running test:
http://crocodile.qa.suse.cz/tests/3201

### CODE REVIEW

File: products/sle/main.pm [hunk 1]
```diff
@@ -658,6 +658,7 @@ sub load_consoletests() {
             }
             elsif (!get_var("ZDUP") and !check_var('VERSION', '12')) {    # zypper and sle12 doesn't do upgrade or installation snapshots
                 loadtest "console/installation_snapshots.pm";
+                loadtest "console/snapper_cleanup.pm";
```
Comment (okurz) [inline products/sle/main.pm]: what about a corresponding TW test? Please check if it works with a current TW snapshot and then add to the main.pm there

Comment (dgutu-suse) [inline products/sle/main.pm]: I didn't checked it against TW, sorry. Let me check it.

File: tests/console/snapper_cleanup.pm [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    script_run("df -h /");
+    script_run("btrfs filesystem df -h /");
+    #4
+    assert_script_run("$snapper_create", 800);
+    script_run("btrfs qgroup show -p / ", 3);
+    script_run("snapper cleanup number");    # cleanup created snapshots
+    assert_script_run("btrfs qgroup show -p / ", 3);
+    script_run("snapper ls");
+    clear_console;
```
Comment (okurz) [inline tests/console/snapper_cleanup.pm]: [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself). You are calling the same code multiple times, please [extract method](http://refactoring.com/catalog/extractMethod.html).

**Comment by okurz** [comment]
* see inline comments
* [squash commits](https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history)
* test for TW

**Comment by sysrich** [comment]
Tagging as 'question' because of the open question regarding Tumbleweed

Tagging as 'WIP' because I agree with Oliver that there is room for cleaner code in the snapper_cleanup.pm

**Comment by dgutu-suse** [comment]
Last test before squash - http://crocodile.qa.suse.cz/tests/3294
Opinion?

**Comment by okurz** [comment]
opinion on what? I don't see a change in operation. I worked already perfect in before. We were just asking to squash the commits so that the two "Perltidy" become one *and* DRY, but no change in functionality :-)

**Comment by dgutu-suse** [comment]
I didn't change nothing in the functionality, IMO now the code look better.

**Comment by dgutu-suse** [comment]
:-1:  can't squash anymore, closing as bad experience.

**Comment by okurz** [comment]
read git docu, you can actually recover from this situation, I am sure and I believe in you :-) You could for example call `git reset --soft origin/master` to delete all commits but keeping your changes in place and then recreate the git commit as you like. It is way easier for the reviewers to see what you did by keeping the original PR in place.

**Pull request closed without merging**